src/engine/macros.rs

```diff
@@ -23,10 +23,14 @@ macro_rules! destroy_filesystems {
     ( $s:ident; $fs:expr ) => {
         let mut removed = Vec::new();
         for uuid in $fs.iter().map(|x| *x) {
+            match $s.filesystems.get_mut_by_uuid(&uuid) {
```

Comment:
Possibly there is something that should be done about the fact that the return value is immutable. I will take a look.